src/main/java/com/marklogic/kafka/connect/source/JsonRowBatcherSupplier.java

```diff
@@ -0,0 +1,59 @@
+package com.marklogic.kafka.connect.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class JsonRowBatcherSupplier extends RowBatcherSupplier {
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+    private final Map<String, Object> parsedConfig;
+    private final List<SourceRecord> sourceRecords;
+
+    JsonRowBatcherSupplier(DataMovementManager dataMovementManager, List<SourceRecord> sourceRecords, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.sourceRecords = sourceRecords;
+        this.parsedConfig = parsedConfig;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+    }
+
+    public RowBatcher<JsonNode> get() {
+        ContentHandle<JsonNode> contentHandle = new JacksonHandle().withFormat(Format.JSON).withMimetype("application/json");
+        RowBatcher<JsonNode> rowBatcher =  dataMovementManager.newRowBatcher(contentHandle);
+        configureRowBatcher(parsedConfig, rowBatcher);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, sourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<?> event, List<SourceRecord> newSourceRecords) {
+        @SuppressWarnings("unchecked") com.fasterxml.jackson.databind.JsonNode rows = ((RowBatchSuccessListener.RowBatchResponseEvent<com.fasterxml.jackson.databind.JsonNode>) event).getRowsDoc().get("rows");
```

Comment:
You can avoid this casting by explicitly declaring `RowBatcherResponseEvent<JsonNode>` in the method signature.